nixos/modules/services/web-servers/keter/config.nix

```diff
@@ -0,0 +1,87 @@
+{ globalketerConfig
+, keterRoot
+, writeShellScript
+, writeTextFile
+, lib
+, bundleCfg
+, callPackage
+, coreutils
+, keter
+, ...
+}:
+
+let
+  incoming = "${keterRoot}/incoming";
+
+
+  globalKeterConfigFile = writeTextFile {
+    name = "keter-config.yml";
+    text = (lib.generators.toYAML { } globalketerConfig);
+  };
+
+  # If things are expected to change often, put it in the bundle!
+  bundle = callPackage ./bundle.nix
+    (bundleCfg // { inherit executable; });
+
+  # This indirection is required to ensure the nix path
+  # gets copied over to the target machine in remote deployments.
+  # Furthermore, it's important that we use exec to
+  # run the binary otherwise we get process leakage.
+  executable = writeShellScript "bundle-wrapper" ''
+    #!/usr/bin/env bash
+    export PATH=$PATH:${lib.makeBinPath [ coreutils ]}
+    set -e
+    ${bundleCfg.secretScript}
+    set -xe
+    ${bundleCfg.publicScript}
+    exec ${bundleCfg.executable}
+  '';
+
+in
+{
+  # This is keter, which loads our app on deploy, and checks if it is healthy
+  # be very careful putting anything in here that changes,
+  # any change to the systemd unit can cause brief windows of downtime
+  # because nixos has to restart the unit.
+  keter = {
+    description = "keter app loader";
+    script = ''
+      set -xe
+      mkdir -p ${incoming}
+      { tail -F /opt/keter/log/keter/current.log -n 0 & ${keter}/bin/keter ${globalKeterConfigFile}; }
+    '';
+    wantedBy = [ "multi-user.target" "nginx.service" ];
+
+    serviceConfig = {
+      Restart = "always";
+      RestartSec = "10s";
+    };
+
+    after = [
+      "network.target"
+      "local-fs.target"
+      "postgresql.service"
+      "redis.service"
```

Comment:
Which Redis? The Redis NixOS module supports multiple instances, go look at that, and add an instance to this module.

Reply:
good point, the app being loaded may need redis, but keter itself doesn't, I removed it for now